add sender email and its password to environment variable

AhmedMaher309 opened the pull request.

AhmedMaher309 committed:
add sender email and its password to environment variable

AhmedMaher309 committed:
Update readme.md

ammaralmorsi commented on readme.md:
````diff
@@ -34,12 +34,18 @@ pip install -r requirements.txt
+5. Add the following environment variables:
+    
+```bash
+export EMAIL=<The email needed to send verification>
````
```suggestion
export EVENTBRITE_EMAIL=<The email needed to send verification>
```
This avoids any conflict because an EMAIL environment variable may be used by the user for another reasons. It is better to give context as possible.

ammaralmorsi commented on readme.md:
````diff
@@ -34,12 +34,18 @@ pip install -r requirements.txt
+    
+```bash
+export EMAIL=<The email needed to send verification>
+export PASSWORD=<its password>
````
```suggestion
export EVENTBRITE_PASSWORD=<its password>
```
This avoids any conflict because an PASSWORD environment variable may be used by the user for another reasons. It is better to give context as possible.

ammaralmorsi commented on routers/auth/commons.py:
```diff
@@ -93,7 +94,7 @@ def send_verification_email(email: str, token: str):
     # Create a message object
     message = MIMEMultipart()
-    message["From"] = "a7medmaher309@gmail.com"
+    message["From"] = os.environ.get("EMAIL")
```
```suggestion
    message["From"] = os.environ.get("EVENTBRITE_EMAIL")
```

ammaralmorsi commented on routers/auth/commons.py:
```diff
@@ -106,8 +107,8 @@ def send_verification_email(email: str, token: str):
         server.starttls()
-        server.login("a7medmaher309@gmail.com", "gsrbcwieakzdxyvk")
-        server.sendmail("your_email@example.com", email, message.as_string())
+        server.login(os.environ.get("EMAIL"), os.environ.get("PASSWORD"))
```
```suggestion
        server.login(os.environ.get("EVENTBRITE_EMAIL"), os.environ.get("EVENTBRITE_PASSWORD"))
```

ammaralmorsi commented on routers/auth/commons.py:
```diff
@@ -93,7 +94,7 @@ def send_verification_email(email: str, token: str):
     """
     # Create a message object
```
I think it will be better if we make a variable called <source_email> and assign it the value of the environment variable instead of every time calling os.environ.get("<variable_name>"), this is error prone and it will be clean if we use a variable at the top of the function.
The same should be done send_forget_password_email function.

AhmedMaher309 committed:
add the settings needed to configure the email needed for verification in environment variables  in the readme file

AhmedMaher309 committed:
add the verification email configuration in  the readme file

AhmedMaher309 committed:
changes commons.py and auth.py in auth

1-changed the verification functions to use the EVENTBRITE_EMAIL and EVENTBRITE_PASSWORD from the environment variables as source_email and source_password

2- updates the change password api to hash the password before updating it in the database

ammaralmorsi commented:
move source email and its password to environment variables.
